packages/test/test-end-to-end-tests/src/test/SummarizeFetchValidation.spec.ts

```diff
@@ -223,58 +234,71 @@ describeNoCompat("Summarizer fetches expected number of times",
+        assert(versionWrap1.fetchCount === 0, "No fetch should have happened");
+        assert(versionWrap1.summaryVersion, "Summary version should be defined");
         summarizerClient.close();
 
         // Add more summaries and we can have more recent ones.
         const secondSummarizer = await createSummarizer(provider, mainContainer);
-        versionWrap = await incrementCellValueAndRunSummary(secondSummarizer, 2 /* expectedMatrixCellValue */);
-
-        // Only ODSP driver uses cache
-        const isUsingCache = provider.driver.type === "odsp";
+        const versionWrap2 = await incrementCellValueAndRunSummary(secondSummarizer, 2 /* expectedMatrixCellValue */);
+        // Only ODSP driver uses snapshot cache due to which the summarizer would have loaded from an older summary.
+        if (provider.driver.type === "odsp") {
+            assert(versionWrap2.fetchCount === 1, "Fetch should have happened");
+            assert.strictEqual(
+                versionWrap2.fetchSnapshotRefSeq, versionWrap1.summaryRefSeq, "Fetch did not download latest snapshot");
+        } else {
+            assert(versionWrap2.fetchCount === 0, "Fetch should have happened");
+        }
 
-        assert(versionWrap.fetchCount === (isUsingCache ? 1 : 0), "Fetch should have happened when using cache");
-        versionWrap = await incrementCellValueAndRunSummary(secondSummarizer, 3 /* expectedMatrixCellValue */);
-        assert(versionWrap.fetchCount === 0, "No fetch should have happened");
+        const versionWrap3 = await incrementCellValueAndRunSummary(secondSummarizer, 3 /* expectedMatrixCellValue */);
+        assert(versionWrap3.fetchCount === 0, "No fetch should have happened");
         await provider.ensureSynchronized();
         secondSummarizer.close();
 
         // Load summarizer from previous version triggers fetch.
         const newSummarizerClient = await createSummarizer(provider, mainContainer);
-        versionWrap = await incrementCellValueAndRunSummary(newSummarizerClient, 4 /* expectedMatrixCellValue */);
-
-        assert(versionWrap.fetchCount === (isUsingCache ? 1 : 0), "Single fetch should have happened once");
-        assert(versionWrap.summaryVersion, "Summarizer should have happened");
+        const versionWrap4 = await incrementCellValueAndRunSummary(newSummarizerClient, 4 /* expectedMatrixCellValue */);
+        assert(versionWrap4.summaryVersion, "Summarizer should have happened");
+        // Only ODSP driver uses snapshot cache due to which the summarizer would have loaded from an older summary.
+        if (provider.driver.type === "odsp") {
+            assert(versionWrap4.fetchCount === 1, "Fetch should have happened");
+            assert.strictEqual(
+                versionWrap4.fetchSnapshotRefSeq, versionWrap3.summaryRefSeq, "Fetch did not download latest snapshot");
+        } else {
+            assert(versionWrap4.fetchCount === 0, "Fetch should have happened");
```

Comment:
what's the difference between fetchCount ===1 and fetchcount ===0? Fetch should have happened in both scenarios?

Reply:
Nope. In case of local server, there is no snapshot cache so it always loads from the latest summary. There will be no referesh from snapshot happening in these cases.